jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,228 @@ def __init__(self, input_models, reffiles=None, **pars):
+        # How many unique values of detector?
+        uq_det = np.unique(detector)
+        ndet = len(uq_det)
+        for idet in range(ndet):
+            indx = (np.where(detector == uq_det[idet]))[0]
+            ndet_files = int(len(indx))
+            self.flag_outliers(idet, uq_det, ndet_files,
+                               diffaxis, nx, ny,
+                               kern_size, threshold_percent,
+                               save_intermediate_results,
+                               ifu_second_check)
+
+        # send input_models back to outlier_detection.py.
+        # self.input_moels is  that is what is returned from outlier_detection.py
+        self.detect_outliers_ifu(self.input_models)
+
+    def flag_outliers(self, idet, uq_det, ndet_files,
+                      diffaxis, nx, ny,
+                      kern_size, threshold_percent,
+                      save_intermediate_results,
+                      ifu_second_check):
+        """Flag outlier pixels in DQ of input images."""
+
+        # set up array to hold group differences
+        diffarr = np.zeros([ndet_files, ny, nx])
+        j = 0
+        for i, model in enumerate(self.input_models):
+            sci = model.data
+            dq = model.dq
+            detector = model.meta.instrument.detector.lower()
+            # only use data from the same detector
+            if detector == uq_det[idet]:
+                bad = np.where(np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool))
+                # set all science data that have DO_NOT_USE to NAN
+                sci[bad] = np.nan
+
+                # Compute left and right differences (MIRI dispersion axis = 1)
+                # For NIRSpec dispersion axis = 0, these differences are top, bottom
+                # prepend = 0 has the effect of keeping the same shape as sci and
+                # for MIRI data (disp axis = 1) the first column = sci data
+                # OR
+                # for NIRSpec data (disp axis = 0) the first row = sci data
+
+                leftdiff = np.diff(sci, axis=diffaxis, prepend=0)
+                flip = np.flip(sci, axis=diffaxis)
+                rightdiff = np.diff(flip, axis=diffaxis, prepend=0)
+                rightdiff = np.flip(rightdiff, axis=diffaxis)
+
+                # Combine left and right differences with minimum of the abs value
+                # to avoid artifacts from bright edges
+                comb = np.zeros([2, ny, nx])
+                comb[0, :, :] = np.abs(leftdiff)
+                comb[1, :, :] = np.abs(rightdiff)
+                combdiff = np.nanmin(comb, axis=0)
+                diffarr[j, :, :] = combdiff
+                j = j + 1
+
+        # minarr final minimum combined differences, size: ny X nx
+        minarr = np.nanmin(diffarr, axis=0)
+        # store where the minarr is nan (neighbor pixels have nan so differences produces a nan)
+        nanminarr = np.isnan(minarr)
+        nanindx = np.where(nanminarr)
```

Comment:
I removed this one nanindx - it is used if 'if_second_check ' is True

Reply:
@nden see if the use of nanindx now ok. I have np.where because the location of nans is used in other arrays. Maybe there is a smarter way to do this without np.where